### PR TITLE
feat: add semi-transparent overlay to QRScanner

### PR DESCRIPTION
**Pull request by mirceanis**
## **Description**

This PR is part of a set of changes intended to redesign the QRScanner experience (see https://github.com/MetaMask/MetaMask-planning/issues/2384)

## **Related issues**

Closes:

* https://github.com/MetaMask/MetaMask-planning/issues/2566
* https://github.com/MetaMask/MetaMask-planning/issues/2567
* https://github.com/MetaMask/MetaMask-planning/issues/2568
* https://github.com/MetaMask/MetaMask-planning/issues/2569

## **Manual testing steps**

1. Go to the home screen and press the scan button
2. The QR scanning camera now has a semi-transparent overlay surrounding the frame, with updated labels

### **Before**
before

### **After**

after

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by mirceanis** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/components/Views/QRScanner/styles.ts [hunk 1]
```diff
@@ -20,13 +25,29 @@ const createStyles = (theme: Theme) =>
...
+      backgroundColor: colors.overlay,
+    },
     frame: {
       width: 250,
       height: 250,
       alignSelf: 'center',
       justifyContent: 'center',
-      marginTop: 100,
-      opacity: 0.5,
+      margin: -4,
```
Comment (jonybur) [inline app/components/Views/QRScanner/styles.ts:50]: Let's try to avoid negative margins

**Review by jonybur — CHANGES_REQUESTED** [review]

File: app/components/Views/QRScanner/styles.ts (hunk 1 above)
Comment (mirceanis) [reply app/components/Views/QRScanner/styles.ts:50]: I don't know how else to overlap the frame with the overlay elements. Any suggestions?

File: app/styles/common.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ export const colors = {
   blackTransparent: 'rgba(0, 0, 0, 0.5)',
   whiteTransparent: 'rgba(255, 255, 255, 0.7)',
   transparent: 'transparent',
+  overlay: 'rgba(242, 244, 246, 0.33)',
+  overlayText: 'rgba(20, 22, 24, 1.0)',
```
Comment (mirceanis) [inline app/styles/common.ts:15]: not sure if this is the right way to introduce colors @jonybur

File: app/components/Views/QRScanner/styles.ts [hunk 3]
```diff
@@ -18,23 +23,40 @@ const createStyles = (theme: Theme) =>
...
+      flexBasis: 0,
+      backgroundColor: colors.overlay,
+    },
     frame: {
       width: 250,
       height: 250,
       alignSelf: 'center',
       justifyContent: 'center',
-      marginTop: 100,
-      opacity: 0.5,
```
Comment (mirceanis) [inline app/components/Views/QRScanner/styles.ts:29]: removing this opacity fixes https://github.com/MetaMask/MetaMask-planning/issues/2569
we already have the frame asset

**Comment by mirceanis** [comment]
changes were cherry picked in https://github.com/MetaMask/metamask-mobile/pull/10061

**Pull request closed without merging**